packages/Button/src/Button.styles.js

```diff
@@ -80,37 +98,20 @@ const disabledTextStyles = css`
 
   &, &:hover {
     background: none;
-    border-color: transparent;
   }
 `;
 
 export const activeStyles = css`
   ${stylers.focusRing.bordered()}
 `;
 
-export const inactiveStyles = ({ kind }) => css`
-  ${borderStyles}
-
-  [data-whatinput="mouse"] &:not([data-has-forced-focus="true"]):focus {
-    ${enabled(css`
-      ${borderStyles}
-
-      ${[types.FLAT, types.MINOR, types.LINK].includes(kind)
-        ? css`
-            box-shadow: none;
-          `
-        : css`
-            box-shadow: ${dropShadow};
-          `}
-    `)}
-  }
```

Comment:
I think this was the actual problematic block for `v5`.  I'm still not actually sure why it would work in `v4` and not `v5`. 

Most of the rest of the changes in this PR are to make it easier to read, diagnose, and maintain this code.